src/dev/amdgpu/pm4_packet_processor.cc

```diff
@@ -586,6 +589,47 @@ PM4PacketProcessor::updateReadIndex(Addr offset, uint64_t rd_idx)
     queuesMap[offset]->getMQD()->mqdReadIndex = rd_idx;
 }
 
+void
+PM4PacketProcessor::unmapAllQueues(bool unmap_static)
+{
+    auto &hsa_pp = gpuDevice->CP()->hsaPacketProc();
+    for (auto iter : gpuDevice->getUsedVMIDs()) {
+        for (auto id : iter.second) {
+            assert(queues.count(id));
+
+            // Do not unmap KMD queues.
+            if (queues[id]->privileged()) {
+                continue;
+            }
+
+            // Do not unmap static queues if requested.
```

Comment:
@v-ramadas : it has nothing to do with this patch, but is this the code you've been looking for?